src/s2python/authorization/client.py

```diff
@@ -1,67 +1,361 @@
-from abc import ABC, abstractmethod
-from typing import Any, Dict
+"""
+S2 protocol client for handling pairing and secure connections.
+"""
 
+import abc
+import json
+import uuid
+import datetime
+from typing import Dict, Optional, Tuple, Union, List, Any
 
-class AbstractConnectionClient(ABC):
-    """Abstract class for handling the /requestConnection endpoint."""
 
-    def request_connection(self) -> Any:
-        """Orchestrate the connection request flow: build → execute → handle."""
-        request_data = self.build_connection_request()
-        response_data = self.execute_connection_request(request_data)
-        return self.handle_connection_response(response_data)
+from jwskate import Jwk
+from pydantic import BaseModel
+
+from s2python.generated.gen_s2_pairing import (
+    ConnectionDetails,
+    ConnectionRequest,
+    PairingRequest,
+    PairingResponse,
+    PairingToken,
+    S2NodeDescription,
+    Protocols,
+)
+
+
+REQTEST_TIMEOUT = 10
+PAIRING_TIMEOUT = datetime.timedelta(minutes=5)
+KEY_ALGORITHM = "RSA-OAEP-256"
```

Comment:
I guess we need a discussion on what's the best default and how to communicate this